### PR TITLE
Feature/337 determinant

### PR DESCRIPTION
**Pull request by mtar**
## Description

First implementation of determinant computation using Gaussian elimination with partial pivoting.

Issue/s resolved: #337 

## Changes proposed:
- add linalg.det

## Type of change

- New feature

## Due Diligence

- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/core/linalg/basics.py [hunk 1]
```diff
@@ -40,6 +41,116 @@
...
+        If `a.ndim < 2` or if the length of the last two dimensions is not the same.
+
+    Examples
+    --------
+    >>> a = ht.array([[-2,-1,2],[2,1,4],[-3,3,-1]])
+    >>> ht.linalg.det(a)
+    DNDarray(54., dtype=ht.float64, device=cpu:0, split=None)
+    """
+    # no split in the square matrices
+    if a.split is None or a.split < a.ndim - 2:
```
Comment (coquelin77) [inline heat/core/linalg/basics.py:67]:
```suggestion
    if not a.is_distributed() or a.split < a.ndim - 2:
```
might need to change the error messages to be consistent

**Review by coquelin77 — CHANGES_REQUESTED** [review]
looks great to me. Just a small change

**Comment by coquelin77** [comment]
@mtar i have changed the shape rules to line up with torch's results. can you check this again?

**Comment by mtar** [comment]
@coquelin77 Thanks for addressing the wrong dimension of the result tensor.

I was now able to reduce some lines of code due to a fix in master.

File: heat/core/linalg/basics.py [hunk 2]
```diff
@@ -155,6 +156,90 @@ def cross(
...
+            a.shape[:-2],
+            types.heat_type_of(data),
+            split=sp,
+            device=a.device,
+            comm=a.comm,
+            balanced=a.balanced,
+        )
+
+    if a.ndim < 2:
+        raise RuntimeError("DNDarray must be at least two-dimensional.")
```
Comment (mtar) [inline heat/core/linalg/basics.py:196]: I'm not sure if we should follow the error types of PyTorch. It's RuntimeError mostly.

**Review by coquelin77 — APPROVED** [review]

**Pull request merged**